src/java.desktop/share/classes/com/sun/imageio/plugins/jpeg/JPEGImageReader.java

```diff
@@ -391,6 +391,12 @@ private void checkTablesOnly() throws IOException {
         tablesOnlyChecked = true;
     }
 
+    private void checkNegativeImageIndex(int imageIndex) {
+        if (imageIndex < minIndex) {
+            throw new IndexOutOfBoundsException("imageIndex < " + minIndex);
```

Comment:
Should we irrespectively throw IOBE or check for `seekForwardOnly` before throwing the exception

https://github.com/openjdk/jdk/blob/f423e1d9ad37135abacb8deb2d2151e21768a23e/src/java.desktop/share/classes/javax/imageio/ImageReader.java#L228-L235